client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -163,7 +163,7 @@ var CheckoutThankYou = React.createClass( {
 	 * @returns {*[]} an array of varying size with the component instance, then an optional purchase object possibly followed by a domain name
 	 */
 	getComponentAndPrimaryPurchaseAndDomain: function() {
-		if ( this.isDataLoaded() ) {
+		if ( this.isDataLoaded() && this.props.receiptId ) {
```

Comment:
The best option I can think of here is to just mask the `props.receiptId` check with a method name to try to make the tristate clearer:

```js
isDataLoaded() {
  return ! this.props.receiptId || this.props.receipt.hasLoadedFromServer;
},

isGenericReceipt() {
  return ! this.props.receiptId;
}
```

Which would make the condition in question:

```js
this.isDataLoaded() && ! this.isGenericReceipt()
```